java/vector/src/main/java/org/apache/arrow/vector/complex/FixedSizeListVector.java

```diff
@@ -184,9 +183,16 @@ private void setReaderAndWriterIndex() {
 
   @Override
   public UnionFixedSizeListReader getReader() {
+    if (reader == null) {
+      reader = new UnionFixedSizeListReader(this);
+    }
     return reader;
   }
 
+  public void invalidateReader() {
```

Comment:
This should be `private`

Reply:
Revised. Thanks a lot for your careful review.